modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -234,3 +280,74 @@ def _read(cls, filepath_or_buffer, **kwargs):
         if index_col is None:
             new_query_compiler._modin_frame._apply_index_objs(axis=0)
         return new_query_compiler
+
+    @classmethod
+    def _define_header_size(
+        cls,
+        header: Union[int, Sequence[int], str, None] = "infer",
+        names: Union[Sequence, None] = None,
+    ):
+        """
+        Defines the number of rows that are used by header.
+
+        Parameters
+        ----------
+        header: int, list of int or str ("infer")
+                original header parameter of read_csv function
+        names:  array
+                original names parameter of read_csv function
+
+        Returns
+        -------
+        header_size: the number of rows that are used by header
+        """
+        header_size = 0
+        if header == "infer" and names is None:
+            header_size += 1
+        elif isinstance(header, int):
+            header_size += header + 1
+        elif hasattr(header, "__iter__") and not isinstance(header, str):
+            header_size += max(header) + 1
+
+        return header_size
+
+    @classmethod
+    def _manage_skiprows_parameter(
+        cls,
+        skiprows: Union[int, Sequence[int], Callable, None] = None,
+        header_size: int = 0,
+    ):
+        """
+        Manages read_csv skiprows parameter.
+
+        Parameters
+        ----------
+        skiprows: int, array or callable
+                original skiprows parameter of read_csv function
+        header_size: int
+                number of rows that are used by header
+
+        Returns
+        -------
+        skiprows: updated skiprows parameter
+        monotonic_skiprows: whatever or not skiprows is monotonic range
```

Comment:
I would propose to rename `monotonic_skiprows` somehow. Because `[1, 10, 20]` is a monotonic sequence, but `monotonic_skiprows` will be `False`, that could be confusing while debugging.

Reply:
Ok, `uniform_skiprows` is used.